Sachin_Kumar_prac2 login and verification UI pages

sAchin12320-web opened the pull request:
This contains the login and verification UI pages.

sAchin12320-web committed:
Flutter_resources

This file will contain the resources useful for learning Flutter.

sAchin12320-web committed:
Merge pull request #1 from sAchin12320-web/sAchin12320-web(flutter_resources)

Flutter_resources

sAchin12320-web committed:
Create Task2_Sachin_Kumar

this contains the practice 2 where i have made login and verification ui using flutter.

sAchin12320-web committed:
Add files via upload

sAchin12320-web committed:
Delete Task2_Sachin_Kumar

sAchin12320-web committed:
Delete lib directory

sAchin12320-web committed:
prac2_login_verification_UI

This contains the login and Verification UI made using flutter

sAchin12320-web committed:
Merge pull request #2 from RitwikSingh28/main

Sachin Kumar__ task2 login and verification pages UI created.

sAchin12320-web committed:
Delete prac2_login_verification_UI directory

sAchin12320-web committed:
login and verification pages using flutter.

This consists of login and verification UI's made using flutter

sAchin12320-web committed:
Delete prac2-pages/RitwikSingh directory

sAchin12320-web committed:
Delete .DS_Store

RitwikSingh28 commented:
Please try to provide screenshots of your running app in the description. :thought_balloon:

RitwikSingh28 commented on prac2_login_verification_UI/main.dart:
```diff
@@ -0,0 +1,184 @@
+}
+
+class _LoginpageState extends State<Loginpage> {
+  bool value=true;
```
**Tip**: Whenever you are dealing with instance variables of a class, it is better to follow OOP principles and make them private.
Try to make instance variables private from now on. :sunglasses:

RitwikSingh28 replied:
Resolved in 8a4acbc.... :+1:

RitwikSingh28 commented on prac2_login_verification_UI/main.dart:
```diff
@@ -0,0 +1,184 @@
+                  ),
+                  child: Padding(
+                    padding: EdgeInsets.fromLTRB(20.0,0,0,0),
+                    child: TextField(
```
We would like to hide away the text input by user in a password field, we should see *** instead of 'abc'.
Please make this fix in subsequent commits :100:

RitwikSingh28 replied:
Resolved in 65e2e0c.... :+1:

RitwikSingh28 commented on prac2_login_verification_UI/main.dart:
```diff
@@ -0,0 +1,184 @@
+  bool value=true;
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
```
Please try to resolve this issue. (bottom overflow)
**You may have to see how to remove overflow when keypad opens up.**

![WhatsApp Image 2022-12-23 at 10 45 42](https://user-images.githubusercontent.com/106133014/209275580-dc5075a1-4c39-4546-91bb-4738bb6d5f9e.jpeg)

RitwikSingh28 replied:
Resolved in a3c6fdf... :+1:

RitwikSingh28 commented on prac2_login_verification_UI/main.dart:
```diff
@@ -0,0 +1,184 @@
+                  ),
+                  child: Padding(
+                    padding: EdgeInsets.fromLTRB(20.0,0,0,0),
+                    child: TextField(
```
When the textfield is in focus, we see a blue line underneath. Please try to remove it. :exclamation: 

If you want to show that the textfield is selected, you can instead darken the borders of the field...

RitwikSingh28 replied:
Resolved in 7d8b71d... :+1:

RitwikSingh28 commented on prac2_login_verification_UI/verification.dart:
```diff
@@ -0,0 +1,202 @@
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: Padding(
```
Consider wrapping the body in a **SafeArea** widget.
Please refer to the documentation to get an idea of what it is and how to use it.

RitwikSingh28 replied:
Resolved in 4754d2b and 7c24e01... :+1:

RitwikSingh28 commented on prac2_login_verification_UI/verification.dart:
```diff
@@ -0,0 +1,202 @@
+import 'package:flutter/material.dart';
```
This page also suffers from overflowing at the bottom. Please try to resolve it too.

You should look at ScreenUtil package in flutter.

![WhatsApp Image 2022-12-23 at 10 45 41](https://user-images.githubusercontent.com/106133014/209276058-02edf91a-cbad-4306-9422-8f8b2a34af21.jpeg)

RitwikSingh28 replied:
The problem still persists @sAchin12320-web even after your minor UI adjustments.

Please consider using ScreenUtil package to make the screen adaptive to various display sizes. :thought_balloon:

RitwikSingh28 reviewed (changes requested):
Nice work so far @sAchin12320-web !! :sparkles: 

Please try to solve the issues outlined in the comments in subsequent commits.

**Note**:
Try to resolve one problem in a single commit. And another in a different commit.
It is a good practice to do so. :thought_balloon:

sAchin12320-web committed:
Private instance variables

I have made the instance variables private using '_'.

sAchin12320-web committed:
Password issue resolved 

Using obscureText , the textarea having password is not showing the character entered as before.

sAchin12320-web committed:
Keypad overflow issue

The issue caused when keypad pops out is resolved using book resizeToAvoidBottomInset

sAchin12320-web committed:
Wrapping body under SafeArea widget

From this, i get in through the benefits of using SafeArea widget.

sAchin12320-web committed:
Wrapping body inside SafeArea widget

I have wrapped the body with SafeArea widget as per the issue generated.

sAchin12320-web committed:
Pixel overflow resolved.

I have used the property resizeToAvoidBottomInset to cope up with the issue.

sAchin12320-web committed:
Resolved the issue of blue underline in textfield

The blue underline comes whenever the textfield is focussed. Using border:InputBorder.none in decoration block solve the issue occurred.

RitwikSingh28 reviewed (changes requested):
Please look through the remaining issues and try to solve them! :exclamation:
